tools/src/h5dump/h5dump_xml.c

```diff
@@ -3446,12 +3548,20 @@ xml_dump_fill_value(hid_t dcpl, hid_t type)
     ctx.indent_level++;
     dump_indent += COL;
 
-    space = H5Tget_size(type);
+    H5E_BEGIN_TRY
+    {
+        n_type = H5Tget_native_type(type, H5T_DIR_DEFAULT);
```

Comment:
These changes were copied from the DDL output code and fix an issue with XML output where the native type wasn't being used when retrieving the fill value of a dataset. The values displayed would be wrong when dumping a dataset with a different endianness than the machine, for example.